From b763ae2fd0087ab99c76bc1947ed4e098c032446 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Mon, 14 Mar 2016 10:36:58 +0200
Subject: [PATCH 308/666] crypto: testmgr - add another test for
 rfc4543(gcm(aes))
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This test is intended to check if implementations behave correctly
when there is no need for padding - since data to be authenticated
is already 16B (AES BLOCK SIZE) aligned:
AAD + IV + Payload = 8 + 8 + 64 = 80 bytes

Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 crypto/testmgr.h | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/crypto/testmgr.h b/crypto/testmgr.h
index 4293573..5a74c17 100644
--- a/crypto/testmgr.h
+++ b/crypto/testmgr.h
@@ -22391,6 +22391,34 @@ static const struct aead_testvec aes_gcm_rfc4543_enc_tv_template[] = {
 			  "\xe1\x55\x10\x6a\xa8\xdc\xd6\x18"
 			  "\xe4\x09\x9a\xaa",
 		.rlen	= 68,
+	}, {
+		.key	= "\xfe\xff\xe9\x92\x86\x65\x73\x1c"
+			  "\x6d\x6a\x8f\x94\x67\x30\x83\x08"
+			  "\xca\xfe\xba\xbe",
+		.klen	= 20,
+		.iv	= "\xfa\xce\xdb\xad\xde\xca\xf8\x88",
+		.assoc	= "\x00\x00\xa5\xf8\x00\x00\x00\x0a",
+		.alen	= 8,
+		.input	= "\x45\x00\x00\x3e\x69\x8f\x00\x00"
+			  "\x80\x11\x4d\xcc\xc0\xa8\x01\x02"
+			  "\xc0\xa8\x01\x01\x0a\x98\x00\x35"
+			  "\x00\x2a\x23\x43\xb2\xd0\x01\x00"
+			  "\x00\x01\x00\x00\x00\x00\x00\x00"
+			  "\x03\x73\x69\x70\x09\x63\x79\x62"
+			  "\x65\x72\x63\x69\x74\x79\x02\x64"
+			  "\x6b\x00\x00\x01\x00\x01\x00\x01",
+		.ilen	= 64,
+		.result	= "\x45\x00\x00\x3e\x69\x8f\x00\x00"
+			  "\x80\x11\x4d\xcc\xc0\xa8\x01\x02"
+			  "\xc0\xa8\x01\x01\x0a\x98\x00\x35"
+			  "\x00\x2a\x23\x43\xb2\xd0\x01\x00"
+			  "\x00\x01\x00\x00\x00\x00\x00\x00"
+			  "\x03\x73\x69\x70\x09\x63\x79\x62"
+			  "\x65\x72\x63\x69\x74\x79\x02\x64"
+			  "\x6b\x00\x00\x01\x00\x01\x00\x01"
+			  "\x6d\x4a\xe6\x38\x82\x9f\x73\x97"
+			  "\x04\x5f\x7c\x6c\x65\xea\xd0\x2d",
+		.rlen	= 80,
 	}
 };
 
-- 
2.7.4

